cmp/relativetimestamp/RelativeTimestamp.js

```diff
@@ -83,166 +83,71 @@ class LocalModel {
-    opts.isFuture = diff < 0;
-    opts.elapsedTime = Math.abs(diff);
-    return opts;
-};
-
-const normalizeAndValidate = opts => {
-    const {isFuture, allowFuture, nowEpsilon, elapsedTime} = opts;
-    if (elapsedTime < nowEpsilon * SECONDS) {
-        opts.elapsedTime = 0;
-    } else if (!allowFuture && isFuture) {
-        opts.isInvalid = true;
-        console.warn(`Unexpected future date provided for timestamp: ${elapsedTime}ms in the future.`);
+function doFormat(opts) {
+    const diff = opts.relativeTo - opts.timestamp,
+        isFuture = diff < 0,
+        elapsed = Math.abs(diff),
+        suffix = isFuture ? opts.futureSuffix : opts.pastSuffix;
+
+    if (isFuture && !opts.allowFuture) {
+        console.warn(`Unexpected future date provided for timestamp: ${elapsed}ms in the future.`);
+        return '[???]';
     }
 
-    return opts;
-};
-
-const getMillisAndUnit = opts => {
-    const {isInvalid, elapsedTime} = opts;
-    // By default the smallest possible unit should be used
-    opts.unit = 'seconds';
-    opts.millis = 0;
-
-    if (isInvalid || !elapsedTime) return opts;
-
-    const types = [
-        {name: 'seconds',  formula: v => v / SECONDS},
-        {name: 'minute',   formula: v => v / MINUTES},
-        {name: 'hour',     formula: v => v / HOURS},
-        {name: 'day',      formula: v => v / DAYS},
-        {name: 'month',    formula: v => v / (DAYS * 30)},
-        {name: 'year',     formula: v => v / (DAYS * 365)}
-    ];
-
-    types.forEach(type => {
-        const val = type.formula(elapsedTime);
-
-        if (val >= 1) {
-            opts.millis = parseInt(val, 10);
-            opts.unit = type.name;
-
-            if (opts.unit !== 'seconds') {
-                opts.unit = pluralize(opts.unit, opts.millis);
-            }
-        }
-    });
-
-    return opts;
-};
-
-const getSuffix = opts => {
-    const {isInvalid, elapsedTime, isFuture, nowString} = opts;
-    if (isInvalid) return opts;
-
-    if (!elapsedTime && nowString) {
-        opts.suffix = nowString;
-        opts.useNowString = true;
-    } else {
-        opts.suffix = opts[isFuture ? 'futureSuffix' : 'pastSuffix'];
+    if (elapsed < opts.epsilon * SECONDS && opts.equalString) {
+        return opts.equalString;
     }
 
-    return opts;
-};
-
-const getPrefix  = opts => {
-    const {isInvalid, prefix} = opts;
-    if (isInvalid) return opts;
-    opts.prefix =  prefix ? prefix + ' ' : '';
-    return opts;
-};
-
-const getResult = opts => {
-    const {isInvalid, elapsedTime, millis, unit, useNowString, prefix, suffix, short} = opts;
-    if (isInvalid) return '[???]';
-
-    // if elapsedTime was normalized to 0 (smaller than nowEpsilon) then return the nowString
-    // if specified, otherwise return the default FORMAT for seconds.
-    if (!elapsedTime && useNowString) return suffix;
+    let ret = (elapsed > 10 * SECONDS && elapsed < 60 * SECONDS) ?
+        '<1 minute' :  // override weird moment default in this range.
```

Comment:
Not sure what the weird default is, but I wonder if it would be more robust to look for an replace that string instead of keying off of these time ranges?  I suppose both could change with future moment releases, but replacing the weird string might be more robust (and make it more clear what we are swapping out for "<1 minute")

Reply:
The default is 'a few seconds'. This is undesirable, since displaying 'a few seconds' for 40 seconds is strange.

Reply:
1) yep -- that lame comment was mine -- lets fix the comment to specify the problem we have with using a 'few seconds'.

2) Let's extend the start time of that interval, to, say 15 seconds.  Think 'a few seconds' is very nice, and appropriate for slightly larger numbers.  But I strongly agree that it is weird for our users when you get deep into the minute.....

Reply:
@lbwexler You already merged this branch. I have pushed my changes, but I'm not sure what to do with them.